src/main/java/seedu/address/logic/commands/MarkCommand.java

```diff
@@ -62,81 +63,141 @@ private static Task createMarkedTask(Task task) {
+        return result(markedTasks, alreadyMarkedIndexes, outOfBoundsIndexes);
     }
 
     /**
      * Converts the list of successfully marked tasks into a string to be returned to the user
-     * @param unmarkedTasks
+     * @param markedTasks
      * @return
      */
-    private String markedTasksToString(List<Task> unmarkedTasks) {
-        String str = "Uncompleted Tasks: \n";
-        for (int i = 0; i < unmarkedTasks.size(); i++) {
-            str += (i + 1) + ". " + unmarkedTasks.get(i) + "\n";
+    private String markedTasksToString(List<Task> markedTasks) {
+        String str = "Successfully Marked Tasks: \n";
+        for (int i = 0; i < markedTasks.size(); i++) {
+            str += (i + 1) + ". " + markedTasks.get(i) + "\n";
         }
         return str;
     }
 
+    /**
+     * Converts a list of indexes to a string to be returned to the user.
+     * @param indexes
+     * @return
+     */
+    private String indexesToString(List<Index> indexes) {
+        String str = "" + indexes.get(0).getOneBased();
+        for (int i = 1; i < indexes.size(); i++) {
+            int index = indexes.get(i).getOneBased();
+            if (indexes.size() > 1 && i == indexes.size() - 1) {
+                str += " and " + index;
+            } else {
+                str += ", " + index;
+            }
+        }
+        return str;
+    }
+
+
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof MarkCommand // instanceof handles nulls
                 && targetIndexes.equals(((MarkCommand) other).targetIndexes)); // state check
     }
+
+    /**
+     * Checks if index is a valid index to mark.
+     * If index > size of targetIndexes, return -1.
+     * If index is already marked, return 1.
+     * If index is a valid index, return 0.
+     *
+     * @return an int representing validity of the index
+     */
+    private int isValidIndex(Index index, List<Task> taskList) {
+        if (index.getZeroBased() > taskList.size() - 1 || index.getZeroBased() < 0) { //index out of bounds
+            return -1;
+        } else if (taskList.get(index.getZeroBased()).getCompletionStatus().toString().equals("true")) {
+            //already marked index
+            return 1;
+        } else { //valid index
+            return 0;
+        }
+    }
+
+    /**
+     * Processes the lists containing the results of the marking of indexes and returns the result to the user.
+     * @param markedTasks
+     * @param alreadyMarkedIndexes
+     * @param outOfBoundsIndexes
+     * @return CommandResult
+     * @throws CommandException
+     */
+    private CommandResult result(List<Task> markedTasks, List<Index> alreadyMarkedIndexes,
+                                 List<Index> outOfBoundsIndexes) throws CommandException{
+        if (!markedTasks.isEmpty()) { //have successfully marked tasks
```

Comment:
As you are exhausting all possible permutations, would it be possible to reduce the amount of repetition?

I was thinking that it could help to use a `StringBuilder`, that builds an error string if there exists a particular violation (i.e. already marked task exists, or indexes out of bounds exists). In this case, only if there exists a particular violation would you add the error string to the `StringBuilder`. Otherwise, you would do nothing. At the end, you can check if the `StringBuilder` was edited. If it was, it means that there was an error and you can just append the successfully marked tasks and throw the error. 

I think this could help in enforcing the [DRY](https://nus-cs2103-ay2122s2.github.io/website/se-book-adapted/chapters/principles.html#dry-principle) principle and making the [happy path](https://nus-cs2103-ay2122s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) more prominent!

I have included the pseudo-code for this suggestion below:
```
StringBuilder errorString = new StringBuilder();
if (there are already marked tasks)
    errorString.add(already marked error statement)

if (there indexes out of bounds)
    errorString.add(index out of bounds error statement)

if (errorString != "") 
    errorString.add(successfully marked tasks)
    throw CommandException(errorString)

return CommandResult 
```

Reply:
tysm for this suggestion, I was thinking about how I should make the happy path more prominent too!! this makes a lot of sense will take note and update accordingly!! :)